docs/training/apis-howto/api-core-ug.rst

```diff
@@ -1,212 +1,302 @@
+.. code::
 
-#. You can navigate to the experiment in the WebUI and pause it mid-training. The trial shuts down
-   and stops producing logs. If you reactivate training it resumes where it stopped. After training
-   is completed, click Continue Trial to see that fresh training is started but that the model
-   weight continues from where previous training finished.
+   det e create metrics.yaml .
 
-The complete ``2_checkpoints.py`` and ``2_checkpoints.yaml`` listings used in this example can be
-found in the :download:`core_api.tgz </examples/core_api.tgz>` download or in the `Github repository
-<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api>`_.
+Open the Determined WebUI again and navigate to the **Overview** tab.
 
-.. _core-hpsearch:
+The WebUI now displays metrics. In this step, you learned how to add a few new lines of code in
+order to report training and validation metrics to the Determined master. Next, we’ll modify our
+script to report checkpoints.
 
 ***********************
- Hyperparameter Search
+ Step 3: Checkpointing
 ***********************
 
+Checkpointing periodically during training and reporting the checkpoints to the master gives us the
+ability to stop and restart training. In this section, we’ll modify our script for the purpose of
+checkpointing.
+
+In this step, we’ll run our experiment using the ``model_def_checkpoints.py`` script and its
+accompanying ``checkpoints.yaml`` experiment configuration file.
+
+Step 3.1: Save Checkpoints
+==========================
+
+To save checkpoints, add the ``store_path`` function to your script:
+
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_checkpoints.py
+   :language: python
+   :start-after: # Docs snippet start: save checkpoint
+   :end-before: # Docs snippet end: save checkpoint
+   :dedent:
+
+Step 3.2: Continuations
+=======================
+
+There are two types of continuations: pausing and reactivating training using the WebUI or clicking
+**Continue Trial** after the experiment completes.
+
+Each type of continuation has its own behavior. While you always want to preserve the value you are
+incrementing (the “model weight”), you do not always want to preserve the batch index. When you
+pause and reactivate you want training to continue from the same batch index. However, when starting
+a fresh experiment, you want training to start with a fresh batch index.
+
+To distinguish between the two types of continuations, you can save the trial ID in the checkpoint.
+
+**Enable Pausing and Resuming an Experiment**
+
+To enable pausing an experiment, enable preemption:
+
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_checkpoints.py
+   :language: python
+   :start-after: # Docs snippet start: enable preemption
+   :end-before: # Docs snippet end: enable preemption
+   :dedent:
+
+Define a load_state function for restarting model training from existing checkpoint:
```

Comment:
The code shown under here renders with the lines
```
 # Docs snippet start: compare checkpoint and current trial IDs
    # If trial ID does not match our current trial ID, we'll ignore
    # epochs completed and start training from epoch_idx = 0
    if ckpt_trial_id != trial_id:
        epochs_completed = 0
    # Docs snippet end: compare checkpoint and current trial IDs
```
Presumably we don't want the `Docs snippet....` lines in here

Reply:
yes that's not a good experience, i have a docs snippet nested in another docs snippet

Reply:
resolved